packages/core/src/transport/sendWithRetryStrategy.ts

```diff
@@ -0,0 +1,153 @@
+import { addTelemetryDebug } from '@datadog/browser-core'
+import { monitor } from '../tools/monitor'
+import { ONE_KILO_BYTE, ONE_MEGA_BYTE, ONE_SECOND } from '../tools/utils'
+import type { Payload, HttpResponse } from './httpRequest'
+
+export const MAX_ONGOING_BYTES_COUNT = 80 * ONE_KILO_BYTE
+export const MAX_ONGOING_REQUESTS = 32
+export const MAX_QUEUE_BYTES_COUNT = 3 * ONE_MEGA_BYTE
+export const MAX_BACKOFF_TIME = 256 * ONE_SECOND
+
+export interface RetryState {
+  isIntakeAvailable: boolean
+  currentBackoffTime: number
+  bandwidthMonitor: ReturnType<typeof newBandwidthMonitor>
+  queuedPayloads: ReturnType<typeof newPayloadQueue>
+}
```

Comment:
hum, if we want to go this way, I would probably differentiate status this way:
 `INTAKE_UP` | `INTAKE_ERROR_DETECTED` | `INTAKE_DOWN` 
 in order to not switch between `ACCEPTING_REQUESTS` and `QUEUING_REQUESTS` every time we reach a bandwidth limit.
We also need `queuedPayloads` and `bandwidthMonitor` to be shared among all `httpRequestInstance.send`.
So maybe something like:

```ts
interface RetryContext {
  queuedPayloads: ...,
  currentBackoffTime: ...,
  state: RetryState
}

type RetryState = {
  status: INTAKE_UP,
} | {
  status: INTAKE_ERROR_DETECTED,
} | {
  status: INTAKE_DOWN,
  currentBackoffTime: number,
}
```
wdyt?

Reply:
I think that if we need the queue and currentBackoffTime at any time, having separate states is less interesting. Let's keep it your way 🙂

Reply:
`currentBackoffTime` is only needed when intake is down and having named state can ease the understanding.
I'll experiment a bit and see what it could look like.

Reply:
splitting into context and state added quite some complexity without a lot of benefit, I've promoted the boolean into a status, it could make things a bit clearer.